client/tests/conftest.py

```diff
@@ -120,3 +139,74 @@ def azure_blob():
 @pytest.fixture(scope="module")
 def s3(aws_credentials):
     return S3StoreConfig("bucket_path", "bucket_region", aws_credentials)
+
+@pytest.fixture(scope="module")
+def local_provider():
+    ff.register_user("test_user").make_default_owner()
+    return ff.register_local()
+
+@pytest.fixture(scope="module")
+def local_provider_and_source(local_provider):
+    local_source = local_provider.register_file(
+                name="transactions",
+                variant="quickstart",
+                description="A dataset of fraudulent transactions.",
+                path=f"{dir_path}/test_files/input_files/transactions.csv"
+            )
+    return (local_provider, local_source)
+
+@pytest.fixture(scope="module")
+def resource_serving_clients():
+    def get_clients_for_context(is_local):
+            resource_client = ff.ResourceClient(local=True)
+            serving_client = ff.ServingClient(local=True)
+
+            return (resource_client, serving_client)
+    
+    return get_clients_for_context
+
+@pytest.fixture(scope="module")
+def create_temp_file(contents):
+    file = NamedTemporaryFile(delete=False)
```

Comment:
this is cool, I wasn't aware of it. There is also `tmp_path` with pytest (https://docs.pytest.org/en/6.2.x/tmpdir.html)

Reply:
@ahmadnazeri, I can't take credit for this: I simply moved a local fixture from `serving_test.py` into `conftest.py`. I noticed the `tmpdir` fixture in the pytest docs. Should I give that a try to avoid rolling custom solutions to a solved problem?

Reply:
if you have time then that would be great but no worries since `NamedTemporaryFile` is builtin, I believe.